<center>
<h1>Data Analysis using feature treatment, variable analysis Methods</h1> 
 Understanding the dataset features through feature treatment, variable analysis and exploratory analysis.

In [58]:
#1ST SECTION EXPLORATORY DATA ANALYSIS
#importing dataset to explore
import pandas as pd
import numpy as np

file = "Apprentice_Chef_Dataset.xlsx"

chef = pd.read_excel(file)



pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
chef.head(n = 10)

REVENUE  CROSS_SELL_SUCCESS                              NAME                                     EMAIL FIRST_NAME  FAMILY_NAME  TOTAL_MEALS_ORDERED  UNIQUE_MEALS_PURCH  CONTACTS_W_CUSTOMER_SERVICE  PRODUCT_CATEGORIES_VIEWED  AVG_TIME_PER_SITE_VISIT  MOBILE_NUMBER  CANCELLATIONS_BEFORE_NOON  CANCELLATIONS_AFTER_NOON  TASTES_AND_PREFERENCES  PC_LOGINS  MOBILE_LOGINS  WEEKLY_PLAN  EARLY_DELIVERIES  LATE_DELIVERIES  PACKAGE_LOCKER  REFRIGERATED_LOCKER  FOLLOWED_RECOMMENDATIONS_PCT  AVG_PREP_VID_TIME  LARGEST_ORDER_SIZE  MASTER_CLASSES_ATTENDED  MEDIAN_MEAL_RATING  AVG_CLICKS_PER_VISIT  TOTAL_PHOTOS_VIEWED
0   1880.0                   1                      Addam Osgrey                 addam.osgrey@passport.com      Addam       Osgrey                  118                   4                            7                          5                    86.00              1                          2                         1                       0          6              2            8                 0                2               1                    0                            90              165.8                   6                        1                   3                    11                    0
1   1495.0                   1                   Aegon Blackfyre                   aegon.blackfyre@jnj.com      Aegon    Blackfyre                   44                   3                            6                          3                   125.60              1                          0                         1                       0          6              1            8                 0                4               1                    0                            90              150.5                   4                        1                   3                    13                   90
2   2572.5                   1         Aegon Frey (son of Aenys)       aegon.frey.(son.of.aenys)@gmail.com      Aegon        Frey                    38                   1                            5                          3                    58.00              1                          5                         0                       1          6              2           14                 0                0               0                    0                            80               99.6                   3                        0                   3                    12                    0
3   1647.0                   1  Aegon Targaryen (son of Rhaegar)  aegon.targaryen.(son.of.rhaegar)@ibm.com      Aegon   Targaryen                    76                   3                            8                         10                    45.51              0                          3                         0                       1          6              1           11                 5                4               0                    0                            90              125.0                   3                        0                   3                    13                    0
4   1923.0                   1                 Aegon V Targaryen         aegon.v.targaryen@goldmansacs.com      Aegon  V Targaryen                   65                   3                            6                          9                   106.00              1                          1                         1                       0          6              2           12                 0                4               1                    0                            90              135.3                   3                        1                   3                    13                  253
5   1250.0                   1                      Aegor Rivers                    aegor.rivers@gmail.com      Aegor       Rivers                   57                   4                            6                          2                    84.21              1                          0                         0     

Using info() to undestand the complete information of the dataset.

In [3]:
#Checking info of the dataset
chef.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1946 entries, 0 to 1945
Data columns (total 29 columns):
REVENUE                         1946 non-null float64
CROSS_SELL_SUCCESS              1946 non-null int64
NAME                            1946 non-null object
EMAIL                           1946 non-null object
FIRST_NAME                      1946 non-null object
FAMILY_NAME                     1899 non-null object
TOTAL_MEALS_ORDERED             1946 non-null int64
UNIQUE_MEALS_PURCH              1946 non-null int64
CONTACTS_W_CUSTOMER_SERVICE     1946 non-null int64
PRODUCT_CATEGORIES_VIEWED       1946 non-null int64
AVG_TIME_PER_SITE_VISIT         1946 non-null float64
MOBILE_NUMBER                   1946 non-null int64
CANCELLATIONS_BEFORE_NOON       1946 non-null int64
CANCELLATIONS_AFTER_NOON        1946 non-null int64
TASTES_AND_PREFERENCES          1946 non-null int64
PC_LOGINS                       1946 non-null int64
MOBILE_LOGINS                   1946 non-null int64
W

Missing Value Treatment - 
Finding missing value columns to fill clean the data for acquiring accurate results.

In [4]:
#checking missing values count
chef.isnull().any().count()

29

In [8]:
#checking missing values column
chef.isnull().any().sum()

1

We find that from the info() function only the column "FAMILY NAME" has missing values . Since it is an object datatype we fill the values with "NA"

In [9]:
#filling missing values 

fill_family = 'NA'

#imputing it into fill
chef['FAMILY_NAME'] = chef['FAMILY_NAME'].fillna(fill_family)


In [6]:
#Cross checking if any missing values are left out.
chef.isnull().any().sum()


0

Creating a new chef dataset that has only the columns which are more important for our statistical analysis.

In [44]:
#Dropping columns
chef_new = chef.drop(['NAME', 'EMAIL' ,'FIRST_NAME', 'FAMILY_NAME',"CROSS_SELL_SUCCESS"],axis = 1)

In [45]:
#cross checking the columns and entries.
chef_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1946 entries, 0 to 1945
Data columns (total 24 columns):
REVENUE                         1946 non-null float64
TOTAL_MEALS_ORDERED             1946 non-null int64
UNIQUE_MEALS_PURCH              1946 non-null int64
CONTACTS_W_CUSTOMER_SERVICE     1946 non-null int64
PRODUCT_CATEGORIES_VIEWED       1946 non-null int64
AVG_TIME_PER_SITE_VISIT         1946 non-null float64
MOBILE_NUMBER                   1946 non-null int64
CANCELLATIONS_BEFORE_NOON       1946 non-null int64
CANCELLATIONS_AFTER_NOON        1946 non-null int64
TASTES_AND_PREFERENCES          1946 non-null int64
PC_LOGINS                       1946 non-null int64
MOBILE_LOGINS                   1946 non-null int64
WEEKLY_PLAN                     1946 non-null int64
EARLY_DELIVERIES                1946 non-null int64
LATE_DELIVERIES                 1946 non-null int64
PACKAGE_LOCKER                  1946 non-null int64
REFRIGERATED_LOCKER             1946 non-null int64
FOLLO

Using descriptive statistics method for understanding the type of data for its further analysis.

In [46]:
#Using descriptive statistcs trying  to understand the quantitative column, and the type of data it holds.

chef_new.loc[:,:].describe([0.2,
                           0.4,
                           0.6,
                           0.8,
                           0.9,
                           1.0]).round(2)

REVENUE  TOTAL_MEALS_ORDERED  UNIQUE_MEALS_PURCH  CONTACTS_W_CUSTOMER_SERVICE  PRODUCT_CATEGORIES_VIEWED  AVG_TIME_PER_SITE_VISIT  MOBILE_NUMBER  CANCELLATIONS_BEFORE_NOON  CANCELLATIONS_AFTER_NOON  TASTES_AND_PREFERENCES  PC_LOGINS  MOBILE_LOGINS  WEEKLY_PLAN  EARLY_DELIVERIES  LATE_DELIVERIES  PACKAGE_LOCKER  REFRIGERATED_LOCKER  FOLLOWED_RECOMMENDATIONS_PCT  AVG_PREP_VID_TIME  LARGEST_ORDER_SIZE  MASTER_CLASSES_ATTENDED  MEDIAN_MEAL_RATING  AVG_CLICKS_PER_VISIT  TOTAL_PHOTOS_VIEWED
count  1946.00              1946.00              1946.0                      1946.00                    1946.00                  1946.00        1946.00                    1946.00                   1946.00                 1946.00    1946.00        1946.00      1946.00           1946.00          1946.00         1946.00              1946.00                       1946.00            1946.00             1946.00                  1946.00             1946.00               1946.00              1946.00
mean   2107.29                74.63                 4.9                         6.98                       5.38                    99.60           0.88                       1.40                      0.17                    0.71       5.52           1.48        11.33              1.49             2.97            0.36                 0.11                         35.41             150.56                4.44                     0.60                2.79                 13.51               106.43
std    1138.29                55.31                 2.5                         2.28                       3.04                    62.34           0.33                       1.55                      0.43                    0.45       0.58           0.53        13.57              2.32             2.74            0.48                 0.32                         26.58              49.45                1.55                     0.64                0.76                  2.33               181.01
min     131.00                11.00                 1.0                         1.00                       1.00                    10.33           0.00                       0.00                      0.00                    0.00       4.00           0.00         0.00              0.00             0.00            0.00                 0.00                          0.00              33.40                0.00                     0.00                1.00                  5.00                 0.00
20%    1285.00                35.00                 3.0                         5.00                       2.00                    61.58           1.00                       0.00                      0.00                    0.00       5.00           1.00         0.00              0.00             1.00            0.00                 0.00                         10.00             108.60                3.00                     0.00                2.00                 12.00                 0.00
40%    1558.00                51.00                 4.0                         6.00                       4.00                    86.51           1.00                       1.00                      0.00                    1.00       5.00           1.00         5.00              0.00             2.00            0.00                 0.00                         20.00             134.40                4.00                     0.00                3.00                 13.00                 0.00
50%    1740.00                60.00                 5.0                         7.00                       5.00                    94.16           1.00                       1.00                      0.00                    1.00       6.00           1.00         7.00              0.00             2.00            0.00                 0.00                         30.00             145.60                4.00                     1.00                3.00                 13.00                 0.00
60%    1910.00                7

Catogerizing data to understand more clealrly the data type of features we will be handling

In [47]:
"""
# CONTINUOUS OR INTERVAL :
followed a cont distribution[rules-ve inf. to + inf. , assume deciomal place,], logarithmic tranformation.

REVENUE
AVG_TIME_PER_SITE_VISIT
AVG_PREP_VID_TIME
FOLLOWED_RECOMMENDATIONS_PCT

# BINARY(limited to 0 or 1)

MOBILE_NUMBER
TASTES_AND_PREFERENCES
PACKAGE_LOCKER
REFRIGERATED_LOCKER

# COUNT- a different approach to feature engineering, number of items.

TOTAL_MEALS_ORDERED
UNIQUE_MEALS_PURCH
PRODUCT_CATEGORIES_VIEWED
CONTACTS_W_CUSTOMER_SERVICE
CANCELLATIONS_BEFORE_NOON
CANCELLATIONS_AFTER_NOON
PC_LOGINS
MOBILE_LOGINS
WEEKLY_PLAN
EARLY_DELIVERIES
LATE_DELIVERIES
LARGEST_ORDER_SIZE
MASTER_CLASSES_ATTENDED
AVG_CLICKS_PER_VISIT
TOTAL_PHOTOS_VIEWED

# CATEGORICAL- either values.(blue or red),(inside , outside)(more than 2 limits)
MEDIAN_MEAL_RATING


# DISCRETE - very unique, origin, (not very accurate,) 

"""


'\n# CONTINUOUS OR INTERVAL :\nfollowed a cont distribution[rules-ve inf. to + inf. , assume deciomal place,], logarithmic tranformation.\n\nREVENUE\nAVG_TIME_PER_SITE_VISIT\nAVG_PREP_VID_TIME\nFOLLOWED_RECOMMENDATIONS_PCT\n\n# BINARY(limited to 0 or 1)\n\nMOBILE_NUMBER\nTASTES_AND_PREFERENCES\nPACKAGE_LOCKER\nREFRIGERATED_LOCKER\n\n# COUNT- a different approach to feature engineering, number of items.\n\nTOTAL_MEALS_ORDERED\nUNIQUE_MEALS_PURCH\nPRODUCT_CATEGORIES_VIEWED\nCONTACTS_W_CUSTOMER_SERVICE\nCANCELLATIONS_BEFORE_NOON\nCANCELLATIONS_AFTER_NOON\nPC_LOGINS\nMOBILE_LOGINS\nWEEKLY_PLAN\nEARLY_DELIVERIES\nLATE_DELIVERIES\nLARGEST_ORDER_SIZE\nMASTER_CLASSES_ATTENDED\nAVG_CLICKS_PER_VISIT\nTOTAL_PHOTOS_VIEWED\n\n# CATEGORICAL- either values.(blue or red),(inside , outside)(more than 2 limits)\nMEDIAN_MEAL_RATING\n\n\n# DISCRETE - very unique, origin, (not very accurate,) \n\n'

As we can see that for MEDIAN MEAL RATING column, customers are somewhat satisfied with the meal provided.

In [48]:
#### Categorical variable type analysis
print(f"""

MEDIAN_MEAL_RATING
-------------
{chef_new['MEDIAN_MEAL_RATING'].value_counts().sort_index()}
      
""")



MEDIAN_MEAL_RATING
-------------
1      99
2     487
3    1087
4     260
5      13
Name: MEDIAN_MEAL_RATING, dtype: int64
      



From below, it can be clearly seen that most of the customers have provided the mobile number, which could mean a very small percentage is odering through pc logins,and similarly it can be understood that mostly package lockers and refrigirated lockers are not available for many customers.Most customers have already provided information about their tastes and preferences.

In [49]:
#### Binary data
print(f"""

MOBILE_NUMBER
-------------
{chef_new['MOBILE_NUMBER'].value_counts().sort_index()}
      
TASTES_AND_PREFERENCES
-------------
{chef_new['TASTES_AND_PREFERENCES'].value_counts().sort_index()}
      
PACKAGE_LOCKER
-------------
{chef_new['PACKAGE_LOCKER'].value_counts().sort_index()}
           
      
REFRIGERATED_LOCKER
-------------
{chef_new['REFRIGERATED_LOCKER'].value_counts().sort_index()}
      
""")



MOBILE_NUMBER
-------------
0     238
1    1708
Name: MOBILE_NUMBER, dtype: int64
      
TASTES_AND_PREFERENCES
-------------
0     556
1    1390
Name: TASTES_AND_PREFERENCES, dtype: int64
      
PACKAGE_LOCKER
-------------
0    1255
1     691
Name: PACKAGE_LOCKER, dtype: int64
           
      
REFRIGERATED_LOCKER
-------------
0    1726
1     220
Name: REFRIGERATED_LOCKER, dtype: int64
      



In [50]:
import pandas as pd# data science essentials
import seaborn as sns # essential graphical output
import matplotlib.pyplot as plt # enhanced graphical output
import statsmodels.formula.api as smf # regression modeling


# setting pandas print options
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
#not good feature to use -  unique ,prdct viewed,cancellation b4 noon,CANC. AFTER NOON, 


Creating Co-variance matrix to understand the strongest features relevant to the target variable 'REVENUE'

In [51]:
# creating a (Pearson) correlation matrix
df_corr = chef_new.corr().round(2)


# printing (Pearson) correlations with SalePrice
print(df_corr.loc['REVENUE'].sort_values(ascending = False))

REVENUE                         1.00
AVG_PREP_VID_TIME               0.64
MEDIAN_MEAL_RATING              0.61
TOTAL_MEALS_ORDERED             0.60
TOTAL_PHOTOS_VIEWED             0.47
MASTER_CLASSES_ATTENDED         0.45
LARGEST_ORDER_SIZE              0.44
AVG_TIME_PER_SITE_VISIT         0.14
CONTACTS_W_CUSTOMER_SERVICE     0.10
PRODUCT_CATEGORIES_VIEWED       0.03
MOBILE_NUMBER                   0.03
PC_LOGINS                       0.02
CANCELLATIONS_BEFORE_NOON       0.01
TASTES_AND_PREFERENCES          0.01
WEEKLY_PLAN                     0.01
EARLY_DELIVERIES               -0.00
REFRIGERATED_LOCKER            -0.00
LATE_DELIVERIES                -0.01
PACKAGE_LOCKER                 -0.01
FOLLOWED_RECOMMENDATIONS_PCT   -0.02
MOBILE_LOGINS                  -0.02
CANCELLATIONS_AFTER_NOON       -0.04
UNIQUE_MEALS_PURCH             -0.06
AVG_CLICKS_PER_VISIT           -0.55
Name: REVENUE, dtype: float64


Printing all column names  

In [53]:
#Creating a file to print all the columns in df format
chef_new_copy = chef_new.copy()

chef_new_copy = chef_new_copy.drop(['REVENUE'],axis = 1)

for val in chef_new_copy:
    print(f"chef_new['{val}'] +")  

chef_new['TOTAL_MEALS_ORDERED'] +
chef_new['UNIQUE_MEALS_PURCH'] +
chef_new['CONTACTS_W_CUSTOMER_SERVICE'] +
chef_new['PRODUCT_CATEGORIES_VIEWED'] +
chef_new['AVG_TIME_PER_SITE_VISIT'] +
chef_new['MOBILE_NUMBER'] +
chef_new['CANCELLATIONS_BEFORE_NOON'] +
chef_new['CANCELLATIONS_AFTER_NOON'] +
chef_new['TASTES_AND_PREFERENCES'] +
chef_new['PC_LOGINS'] +
chef_new['MOBILE_LOGINS'] +
chef_new['WEEKLY_PLAN'] +
chef_new['EARLY_DELIVERIES'] +
chef_new['LATE_DELIVERIES'] +
chef_new['PACKAGE_LOCKER'] +
chef_new['REFRIGERATED_LOCKER'] +
chef_new['FOLLOWED_RECOMMENDATIONS_PCT'] +
chef_new['AVG_PREP_VID_TIME'] +
chef_new['LARGEST_ORDER_SIZE'] +
chef_new['MASTER_CLASSES_ATTENDED'] +
chef_new['MEDIAN_MEAL_RATING'] +
chef_new['AVG_CLICKS_PER_VISIT'] +
chef_new['TOTAL_PHOTOS_VIEWED'] +


Using statsmodel to output OLS Regression Results and clealry understand which are the most significant feature with respect to the target variable 'Revenue'

In [56]:
# building a base model and trying to understand which are the relevant features tht

# blueprinting a model type
chef_ols_base_model = smf.ols(formula = """chef_new['REVENUE'] ~
                                                chef_new['TOTAL_MEALS_ORDERED'] +
                                                chef_new['UNIQUE_MEALS_PURCH'] +
                                                chef_new['CONTACTS_W_CUSTOMER_SERVICE'] +
                                                chef_new['PRODUCT_CATEGORIES_VIEWED'] +
                                                chef_new['AVG_TIME_PER_SITE_VISIT'] +
                                                chef_new['MOBILE_NUMBER'] +
                                                chef_new['CANCELLATIONS_BEFORE_NOON'] +
                                                chef_new['CANCELLATIONS_AFTER_NOON'] +
                                                chef_new['TASTES_AND_PREFERENCES'] +
                                                chef_new['PC_LOGINS'] +
                                                chef_new['MOBILE_LOGINS'] +
                                                chef_new['WEEKLY_PLAN'] +
                                                chef_new['EARLY_DELIVERIES'] +
                                                chef_new['LATE_DELIVERIES'] +
                                                chef_new['PACKAGE_LOCKER'] +
                                                chef_new['REFRIGERATED_LOCKER'] +
                                                chef_new['FOLLOWED_RECOMMENDATIONS_PCT'] +
                                                chef_new['AVG_PREP_VID_TIME'] +
                                                chef_new['LARGEST_ORDER_SIZE'] +
                                                chef_new['MASTER_CLASSES_ATTENDED'] +
                                                chef_new['MEDIAN_MEAL_RATING'] +
                                                chef_new['AVG_CLICKS_PER_VISIT'] +
                                                chef_new['TOTAL_PHOTOS_VIEWED'] """,
                                                    data = chef_new)


# telling Python to run the data through the blueprint
results = chef_ols_base_model.fit()


# printing the results
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                             OLS Regression Results                            
===============================================================================
Dep. Variable:     chef_new['REVENUE']   R-squared:                       0.640
Model:                             OLS   Adj. R-squared:                  0.636
Method:                  Least Squares   F-statistic:                     148.6
Date:                 Fri, 06 Mar 2020   Prob (F-statistic):               0.00
Time:                         18:51:12   Log-Likelihood:                -15461.
No. Observations:                 1946   AIC:                         3.097e+04
Df Residuals:                     1922   BIC:                         3.110e+04
Df Model:                           23                                         
Covariance Type:             nonrobust                                         
============================================================================================================
                                               coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------------------------
Intercept                                 -183.2421    342.599     -0.535      0.593    -855.147     488.663
chef_new['TOTAL_MEALS_ORDERED']              5.6979      0.353     16.155      0.000       5.006       6.390
chef_new['UNIQUE_MEALS_PURCH']             -63.6897      6.392     -9.965      0.000     -76.225     -51.155
chef_new['CONTACTS_W_CUSTOMER_SERVICE']     45.5299      7.000      6.505      0.000      31.802      59.258
chef_new['PRODUCT_CATEGORIES_VIEWED']        8.8856      5.152      1.725      0.085      -1.218      18.989
chef_new['AVG_TIME_PER_SITE_VISIT']         -0.6782      0.265     -2.556      0.011      -1.199      -0.158
chef_new['MOBILE_NUMBER']                   31.9265     47.925      0.666      0.505     -62.064     125.917
chef_new['CANCELLATIONS_BEFORE_NOON']        0.6012     10.165      0.059      0.953     -19.334      20.537
chef_new['CANCELLATIONS_AFTER_NOON']       -42.9685     36.260     -1.185      0.236    -114.081      28.144
chef_new['TASTES_AND_PREFERENCES']          39.1196     34.679      1.128      0.259     -28.892     107.131
chef_new['PC_LOGINS']                        5.2219     26.948      0.194      0.846     -47.628      58.072
chef_new['MOBILE_LOGINS']                  -36.6814     29.749     -1.233      0.218     -95.025      21.662
chef_new['WEEKLY_PLAN']                     -0.0904      1.155     -0.078      0.938      -2.356       2.175
chef_new['EARLY_DELIVERIES']                 3.2615      6.747      0.483      0.629      -9.972      16.495
chef_new['LATE_DELIVERIES']                  4.5785      5.705      0.802      0.422      -6.611      15.768
chef_new['PACKAGE_LOCKER']                 -22.6944     37.347     -0.608      0.543     -95.940      50.551
chef_new['REFRIGERATED_LOCKER']            -26.1162     56.385     -0.463      0.643    -136.699      84.466
chef_new['FOLLOWED_RECOMMENDATIONS_PCT']    -0.7509      0.593     -1.266      0.206      -1.914       0.413
chef_new['AVG_PREP_VID_TIME']                9.3414      0.632     14.769      0.000       8.101      10.582
chef_new['LARGEST_ORDER_SIZE']             -96.3386     17.066     -5.645      0.000    -129.809     -62.869
chef_new['MASTER_CLASSES_ATTENDED']        175.1745     28.529      6.140      0.000     119.224     231.125
chef_new['MEDIAN_MEAL_RATING']             342.5116     42.105      8.135      0.000     259.936     425.087
chef_new['AVG_CLICKS_PER_VISIT']           -17.8431     13.091     -1.363      0.173     -43.518       7.831
chef_new['TOTAL_PHOTOS_VIEWED']              0.6967      0.099      7.038      0.000       0.503       0.891
==============================================================================
Omnibus:                      694.636   Durbin-Wat

Using OLS Regression resluts and co-variance matrix, we will be creating the best model and use it for finding R-square values

In [65]:
import sklearn.linear_model

Using the stats model we could observe that most relevant features were the ones with p-values lesser than 0.05.

In [66]:
X_features = chef_new[['AVG_TIME_PER_SITE_VISIT','LARGEST_ORDER_SIZE','AVG_CLICKS_PER_VISIT','MASTER_CLASSES_ATTENDED','TOTAL_PHOTOS_VIEWED','MEDIAN_MEAL_RATING','TOTAL_MEALS_ORDERED','UNIQUE_MEALS_PURCH','AVG_PREP_VID_TIME']]
y_target = chef_new[['REVENUE']]

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X_features,
                                                    y_target,
                                                    test_size = 0.25,
                                                    random_state = 222)


In [67]:
from sklearn.linear_model import LinearRegression

# INSTANTIATING a model object
lr = LinearRegression()


# FITTING to the training data
lr_fit = lr.fit(X_features_df , y_target_df)


# PREDICTING on new data
lr_pred = lr_fit.predict(X_test)


# SCORING the results
print('Training Score:', lr.score(X_train, y_train).round(4))
print('Testing Score:',  lr.score(X_test, y_test).round(4))

Training Score: 0.6481
Testing Score: 0.5789


We understood that from the above ols reg base model, the feature [total meals ordered]is relevant to the response or target variable. We observe that R-square value is 0.36 which is not closer to 1, so adding more features will 